### PR TITLE
ZCS-4592:Set zimbraLDAPSchemaVersion early

### PR DESCRIPTION
**Pull request by fatkudu**
On upgrade, was seeing:
    Modifying GLOBALCONFIG globalconfig
      zimbraMtaPostscreenDnsblWhitelistThreshold: 0
    Caught ServiceException while modifying global config attribute zimbraMtaPostscreenDnsblWhitelistThreshold
    com.zimbra.cs.ldap.LdapException$LdapObjectClassViolationException: object class violation - unable to modify attributes: ldap host=zqa-371.eng.zimbra.com:389: object class 'zimbraGlobalConfig' requires attribute 'zimbraLDAPSchemaVersion'
    ExceptionId:main:1521434195320:edd59a1fd07c5007
    Code:ldap.OBJECT_CLASS_VIOLATION
            at com.zimbra.cs.ldap.LdapException.OBJECT_CLASS_VIOLATION(LdapException.java:134)
            at com.zimbra.cs.ldap.unboundid.UBIDLdapException.mapToLdapException(UBIDLdapException.java:60)
            at com.zimbra.cs.ldap.unboundid.UBIDLdapContext.mapToLdapException(UBIDLdapContext.java:239)
            at com.zimbra.cs.ldap.unboundid.UBIDLdapContext.modifyAttributes(UBIDLdapContext.java:434)
            at com.zimbra.cs.ldap.unboundid.UBIDLdapContext.modifyAttributes(UBIDLdapContext.java:398)
            at com.zimbra.cs.account.ldap.ZLdapHelper.modifyAttrs(ZLdapHelper.java:167)
            at com.zimbra.cs.account.ldap.upgrade.UpgradeOp.modifyAttrs(UpgradeOp.java:155)
            at com.zimbra.cs.account.ldap.upgrade.BUG_27075.doEntry(BUG_27075.java:247)
            at com.zimbra.cs.account.ldap.upgrade.BUG_27075.doGlobalConfig(BUG_27075.java:344)
            at com.zimbra.cs.account.ldap.upgrade.BUG_27075.doUpgrade(BUG_27075.java:77)
            at com.zimbra.cs.account.ldap.upgrade.LdapUpgrade.upgrade(LdapUpgrade.java:165)
            at com.zimbra.cs.account.ldap.upgrade.LdapUpgrade.main(LdapUpgrade.java:180)
    Caused by: LDAPException(resultCode=65 (object class violation), errorMessage='object class 'zimbraGlobalConfig' requires attribute 'zimbraLDAPSchemaVersion'', diagnosticMessage='object class 'zimbraGlobalConfig' requires attribute 'zimbraLDAPSchemaVersion'')
            at com.unboundid.ldap.sdk.LDAPConnection.modify(LDAPConnection.java:2442)
            at com.unboundid.ldap.sdk.LDAPConnection.modify(LDAPConnection.java:2383)
            at com.zimbra.cs.ldap.unboundid.UBIDLdapOperation$ModifyAttrs.execute(UBIDLdapOperation.java:468)
            at com.zimbra.cs.ldap.unboundid.UBIDLdapContext.modifyAttributes(UBIDLdapContext.java:404)
            ... 8 more

### CODE REVIEW

File: rpmconf/Install/zmsetup.pl [hunk 1]
```diff
@@ -1394,6 +1394,11 @@ sub installLdapConfig {
 
 sub installLdapSchema {
   main::runAsZimbra("/opt/zimbra/libexec/zmldapschema 2>/dev/null");
+  progress ("Updating zimbraLDAPSchemaVersion to version '$ldapSchemaVersion'\n");
```
Comment (Prashantsurana) [inline rpmconf/Install/zmsetup.pl]: We are already setting the value as default value in zimbra-attrs.xml. 
Can we remove this step completely?

**Comment by fatkudu** [comment]
Tested on Ubuntu 16.  (Saw the same problems reported in zmsetup.log when upgrading from 8.7.9 to build before this fix)

After fix, only mentions of zimbraLDAPSchemaVersion are:
```
sudo grep -n zimbraLDAPSchemaVersion zmsetup.log
11:Tue Mar 20 12:45:46 2018 Updating zimbraLDAPSchemaVersion to version '1521190673'
23:Tue Mar 20 12:45:52 2018 entering function: gcf zimbraLDAPSchemaVersion=1521190673
24:Tue Mar 20 12:45:52 2018 Updating cached global config attribute zimbraLDAPSchemaVersion=1521190673
25:Tue Mar 20 12:45:52 2018 *** Running as zimbra user: /opt/zimbra/bin/zmprov -r -m -l mcf  zimbraLDAPSchemaVersion '1521190673'
```

File: rpmconf/Install/zmsetup.pl (hunk 1 above)
Comment (fatkudu) [reply rpmconf/Install/zmsetup.pl]: My reading is that having specified `requiredIn="globalConfig"` in ldap-attrs.xml, there MUST be a value stored in LDAP, we can't rely on defaulting.

**Comment by fatkudu** [comment]
This change isn't working for a new install :(

**Comment by fatkudu** [comment]
Tested the new change for new install and upgrade and it works.  As expected, the version is updated twice if upgrading.

**Pull request merged**